services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -8068,6 +8068,9 @@ public Map<String, StoreDataAudit> getClusterStaleStores(String clusterName) {
    */
   @Override
   public List<StoreInfo> getStoresForCompaction(String clusterName) {
+    if (compactionManager == null) {
+      throw new VeniceException("CompactionManager is not initialized.");
+    }
```

Comment:
Should we instead assert of `isLogCompactionEnabled` and return `API not supported/enabled`?

Throwing uninitialized could only mean the feature is disabled and returning meaningful response might be helpful to the client.

Reply:
sounds better.